File: bench/dotnet/collatz.cs
namespace Bench;
static class Collatz
{
    public static void Run(int n)
    {
        int best = 0;
        for (int start = 1; start < n; start++)
        {
            long m = start; int steps = 0;
            while (m != 1)
            {
                if (m % 2 == 0) m /= 2; else m = 3 * m + 1;
                steps++;
            }
            if (steps > best) best = steps;
        }
        Console.WriteLine(best);
    }
}

File: bench/python/collatz.py
import os
N = int(os.environ.get("BENCH_N", "500000"))

best = 0
for start in range(1, N):
    n = start
    steps = 0
    while n != 1:
        if n % 2 == 0:
            n = n // 2
        else:
            n = 3 * n + 1
        steps += 1
    if steps > best:
        best = steps

print(best)
